logstash-core/lib/logstash/runner.rb

```diff
@@ -12,129 +13,141 @@
+  option ["--config.reload.interval"], "RELOAD_INTERVAL",
     I18n.t("logstash.runner.flag.reload_interval"),
-    :attribute_name => :reload_interval, :default => 3, &:to_i
+    :attribute_name => "config.reload.interval",
+    :default => LogStash::SETTINGS.get_default("config.reload.interval"), &:to_i
 
-  option ["--http-host"], "WEB_API_HTTP_HOST",
+  option ["--web_api.http.host"], "WEB_API_HTTP_HOST",
     I18n.t("logstash.web_api.flag.http_host"),
-    :attribute_name => :web_api_http_host, :default => "127.0.0.1"
+    :attribute_name => "web_api.http.host",
+    :default => LogStash::SETTINGS.get_default("web_api.http.host")
 
-  option ["--http-port"], "WEB_API_HTTP_PORT",
+  option ["--web_api.http.port"], "WEB_API_HTTP_PORT",
     I18n.t("logstash.web_api.flag.http_port"),
-    :attribute_name => :web_api_http_port, :default => 9600
+    :attribute_name => "web_api.http.port",
+    :default => LogStash::SETTINGS.get_default("web_api.http.port"), &:to_i
 
-  option ["--allow-env"], :flag,
+  option ["--config.allow_env"], :flag,
     I18n.t("logstash.runner.flag.allow-env"),
-    :attribute_name => :allow_env, :default => false
+    :attribute_name => "config.allow_env",
+    :default => LogStash::SETTINGS.get_default("config.allow_env")
 
-  option ["--[no-]log-in-json"], :flag,
+  option ["--log.json"], :flag,
     I18n.t("logstash.runner.flag.log-in-json"),
-    :default => false
-
-  def pipeline_workers=(pipeline_workers_value)
-    @pipeline_settings[:pipeline_workers] = validate_positive_integer(pipeline_workers_value)
-  end
-
-  def pipeline_batch_size=(pipeline_batch_size_value)
-    @pipeline_settings[:pipeline_batch_size] = validate_positive_integer(pipeline_batch_size_value)
-  end
-
-  def pipeline_batch_delay=(pipeline_batch_delay_value)
-    @pipeline_settings[:pipeline_batch_delay] = validate_positive_integer(pipeline_batch_delay_value)
-  end
-
-  def validate_positive_integer(str_arg)
-    int_arg = str_arg.to_i
-    if str_arg !~ /^\d+$/ || int_arg < 1
-      raise ArgumentError, "Expected a positive integer, got '#{str_arg}'"
-    end
-
-    int_arg
-  end
+    :attribute_name => "log.json",
+    :default => LogStash::SETTINGS.get_default("log.json")
 
   attr_reader :agent
 
   def initialize(*args)
     @logger = Cabin::Channel.get(LogStash)
-    @pipeline_settings ||= { :pipeline_id => "main" }
+    @settings = LogStash::SETTINGS
     super(*args)
   end
 
+  def run(args)
+    if i=args.find_index("--settings.path")
+      settings_path = args[i+1]
+    elsif settings_arg = args.find {|v| v.match(/--settings.path=/) }
+      match = settings_arg.match(/--settings.path=(.*)/)
+      settings_path = match[1]
+    elsif ENV['LS_SETTINGS_DIR']
+      settings_path = ENV['LS_SETTINGS_DIR']
+    else
+      settings_path = nil
+    end
+
+    LogStash::SETTINGS.set("settings.path", settings_path) if settings_path
+    LogStash::SETTINGS.from_yaml(LogStash::SETTINGS.get("settings.path"))
```

Comment:
I see the usefulness of having `from_yaml` but could add a `load` method to `LogStash::Settings` that will do the same thing.